lib/middleman-images/extension.rb

```diff
@@ -11,48 +11,37 @@ class Extension < ::Middleman::Extension
       option :cache_dir, 'cache', 'Specification of cache folder'
 
       helpers do
-        def image_tag(url, options = {})
+        def image_tag(path, options = {})
           process_options = options.slice(:resize, :optimize)
-          options = { src: image_path(url, process_options) }.update(options.except(:resize, :optimize))
-          super
+          options = { src: image_path(path, process_options) }.update(options.except(:resize, :optimize))
+          super(path, options)
         end
 
-        def image_path(url, process_options = {})
-          url = extensions[:images].image(url, process_options)
-          super url
+        def image_path(path, process_options = {})
+          path = extensions[:images].image_path(path, process_options)
+          super(path)
         end
       end
 
       def manipulate_resource_list(resources)
-        manipulator.manipulate_resource_list(resources)
+        @manipulator.manipulate_resource_list(resources)
       end
 
-      def template_context
-        @template_context ||= app.template_context_class.new(app, {}, {})
-      end
-
-      def process(source, process_options)
-        destination_path(source, process_options).tap do |dest_url|
-          unless app.sitemap.find_resource_by_path(dest_url)
-            image = Image.new(app, source.source_file, dest_url, process_options.merge(cache_dir: options[:cache_dir]))
-            manipulator.add image
-          end
-        end
-      end
-
-      def image(url, process_options)
-        source = app.sitemap.find_resource_by_path(absolute_image_path(url))
-        return url if source.nil?
+      def image_path(path, process_options)
+        source = app.sitemap.find_resource_by_path(absolute_path(path))
+        return path if source.nil?
 
         process_options[:image_optim] = options[:image_optim]
         process_options[:optimize] = options[:optimize] unless process_options.key?(:optimize)
 
         if process_options[:resize] || process_options[:optimize]
-          url = process(source, process_options)
+          processed_path = add_processed_resource(source, process_options)
+          path = Pathname.new(processed_path).relative_path_from(Pathname.new(app.config[:images_dir])).to_s
```

Comment:
Kann der `Pathname` zum `image_dir` auch in einer Instanzvariable o.ä. (zb `@images_dir_path`) zwischengespeichert werden? Weniger aus Effizienz- denn aus Lesbarkeitsgründen.

Reply:
see cbd8329